context/app/static/js/components/Detail/GlobusLinkMessage/GlobusLinkMessage.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+
+import { LightBlueLink } from 'shared-styles/Links';
+import { StyledExternalLinkIcon } from './style';
+
+const messages = {
+  200: 'Files are available through the Globus Research Data Management System. Access dataset',
+  401: 'Unauthorized access to the Globus Research Management System (bad or expired token). If you believe this to be an error, please contact',
+  403: 'Access to files on the Globus Research Management system are restricted. You may not have access to these files because the Consortium is still curating data and/or the data is protected data that requires you to be a member of the Consortium “Protected Data Group”. Such protected data will be available via dbGaP in the future. If you believe this to be an error, please contact',
+  404: 'Files are not available through the Globus Research Management system. If you believe this to be an error, please contact',
+  500: 'Unexpected server or other error. Report error to',
```

Comment:
Having these fragments of messages that get completed different ways depending on context may be confusing. Possible alternative:
```
if (statusCode === 200) {
   return (
       <>
           {`Files are available through the Globus Research Data Management System. Access dataset ${display_doi} on `}
           <LightBlueLink underline="none" variant="body2" href={url}>
            Globus <StyledExternalLinkIcon />
          </LightBlueLink>
      </>
   )
```
and then handle the errors.

I guess the larger theme is a worry about putting two much logic inside the JSX.

Reply:
I used guard clauses instead of the ternary. Let me know if this is your preference.